core/templates/dev/head/domain/utilities/language-util.service.spec.ts

```diff
@@ -114,11 +114,18 @@ describe('Language util service', function() {
 
   it('should correctly check if language supports autogenerated audio',
     () => {
+      let tryCount = 0;
       // Chrome loads voices asynchronously.
       // https://stackoverflow.com/questions/21513706/
+      // Chrome also calls this multiple times and not all languages are
```

Comment:
That seems weird. Why aren't all languages populated the first time?

Reply:
The API only guarantees that it’s called when the list is altered in some way, so we can’t trust that all languages are populated the first time.

Reply:
Also, where is it calling multiple times? I expect if it is being called multiple times, there is a loop of some sort right?

Reply:
There isn't a loop, but an event that a browser calls then the language list changes. As a result, we might get a hit the first time. In that scenario, the tests will pass as expected. During debugging, I found out that the browser would call this at most twice.

Reply:
Ah I think I see the flow now. Thanks for the clarification.

Reply:
I think all this information in the comments of this thread that you're leaving here is relevant. Could you please add it to the comments in the code as well? Maybe also either reference API links describing exactly how this works, or mention that this was `tested by @kevinlee12` so that people have a reference point in case the behaviour changes in the future.

Reply:
Employed `done` so the conversation above isn't too relevant anymore.

Reply:
Used `done`